pallets/pallet-bonded-coins/src/tests/transactions/refund_account.rs

```diff
@@ -266,6 +266,47 @@ fn refund_below_min_balance() {
 		});
 }
 
+#[test]
+fn refund_account_fails_when_account_blocked() {
+	let pool_details = generate_pool_details(
+		vec![DEFAULT_BONDED_CURRENCY_ID],
+		get_linear_bonding_curve(),
+		false,
+		Some(PoolStatus::Refunding),
+		Some(ACCOUNT_00),
+		None,
+		None,
+		None,
+	);
+	let pool_id: AccountIdOf<Test> = calculate_pool_id(&[DEFAULT_BONDED_CURRENCY_ID]);
+
+	ExtBuilder::default()
+		.with_native_balances(vec![(ACCOUNT_01, ONE_HUNDRED_KILT)])
+		.with_pools(vec![(pool_id.clone(), pool_details)])
+		.with_collaterals(vec![DEFAULT_COLLATERAL_CURRENCY_ID])
+		.with_bonded_balance(vec![
+			(DEFAULT_COLLATERAL_CURRENCY_ID, pool_id.clone(), ONE_HUNDRED_KILT),
+			(DEFAULT_COLLATERAL_CURRENCY_ID, ACCOUNT_01, ONE_HUNDRED_KILT),
+			(DEFAULT_BONDED_CURRENCY_ID, ACCOUNT_01, ONE_HUNDRED_KILT),
+		])
+		.build_and_execute_with_sanity_tests(|| {
+			let origin: OriginFor<Test> = RawOrigin::Signed(ACCOUNT_01).into();
+
+			Assets::block(
+				RawOrigin::Signed(Assets::owner(DEFAULT_COLLATERAL_CURRENCY_ID).unwrap()).into(),
+				DEFAULT_COLLATERAL_CURRENCY_ID,
+				ACCOUNT_01,
+			)
+			.expect("Failed to block account for test");
+
+			// Ensure the refund_account call fails due to pool not being in refunding state
```

Comment:
I guess this comment is outdated, right? This test tests something else.

Reply:
Good catch, I started with a copy of another test case 4e315967d15219f13cb2d3496eee769d42ca987f